src/selectors/per-thread/thread.js

```diff
@@ -132,6 +132,47 @@ export function getThreadSelectorsPerThread(threadIndex: ThreadIndex): * {
     }
   );
 
+  /**
+   * This selector returns the offset to add to a sampleIndex when accessing the
+   * base thread, if your thread is a range filtered thread (all but the base
+   * `getThread` or the last `getPreviewFilteredThread`).
+   */
+  const getSampleIndexOffsetFromCommittedRange: Selector<number> = createSelector(
+    getThread,
+    ProfileSelectors.getCommittedRange,
+    ({ samples }, { start, end }) => {
+      const [beginSampleIndex] = ProfileData.getSampleIndexRangeForSelection(
+        samples,
+        start,
+        end
+      );
+      return beginSampleIndex;
+    }
+  );
+
+  /**
+   * This selector returns the offset to add to a sampleIndex when accessing the
+   * base thread, if your thread is the preview filtered thread.
+   */
+  const getSampleIndexOffsetFromPreviewRange: Selector<number> = createSelector(
+    getFilteredThread,
+    ProfileSelectors.getPreviewSelection,
+    getSampleIndexOffsetFromCommittedRange,
+    ({ samples }, previewSelection, sampleIndexFromCommittedRange) => {
+      if (!previewSelection.hasSelection) {
+        return sampleIndexFromCommittedRange;
+      }
+
+      const [beginSampleIndex] = ProfileData.getSampleIndexRangeForSelection(
+        samples,
+        previewSelection.selectionStart,
+        previewSelection.selectionEnd
+      );
+
+      return sampleIndexFromCommittedRange + beginSampleIndex;
+    }
```

Comment:
Here I hesitated between using the filtered thread (and then I need to sum the `sampleIndexFromCommittedRange`, which means I need to run the selector `getSampleIndexOffsetFromCommittedRange` in addition to this one) (solution A) or using the initial thread directly (solution B).

I chose the solution A because:
1. `getFilteredThread` is probably already memoized, and then running `getSampleIndexRangeForSelection` on a filtered thread is likely faster than on the full thread.
2. while changing the preview selection, the result of `getSampleIndexOffsetFromCommittedRange` would be memoized too.
3. currently `getSampleIndexRangeForSelection` is called twice (once on the thread to create the call tree, and once here), but if we call it with the same arguments we could decide do memoize it (which I didn't implement now because it's not _that_ easy given we call it at 2 locations).

But Solution B could be better once we use bisection for `getSampleIndexRangeForSelection` (draft PR at https://github.com/firefox-devtools/profiler/pull/2170).

Tell me what you think :)